File: client-app/src/models/employee.ts
export interface Employee {
    id: string
    first_name: string
    last_name: string
    position: string
    department: string
}
export interface PositionTrack {
    id: string
    first_name: string
    last_name: string
    position: string
    department: string
    timeStamp: string
    action: string
  }

export interface EmployeeResponse {
    employeeDto: Employee;
    positionTrackDtos: PositionTrack[];
  }

export interface EmployeeUpdate {
    id:string
    first_name: string
    last_name: string
    positionId: string
    departmentId: string
}

export interface EmployeeCreate {
  first_name: string
  last_name: string
  positionId: string
  departmentId: string
}

File: EmployeeManagement.Api/Entities/Employee.cs
namespace EmployeeManagement.Api;

public class Employee
{

    public int Id { get; set; }
    public required string First_name { get; set; }
    public required string Last_name { get; set; }
    public int PositionId { get; set; }
    public int DepartmentId { get; set; }

    public Position? Position { get; set; }
    public Department? Department { get; set; }
    public List<PositionTrack> PositionTrack { get; set; } = [];
}
